### DISCUSSION TITLE
How to add secret with dcli to the vault?

### DISCUSSION THREAD

**Discussion by tsamppa** [Dashlane/dashlane-cli#214] [General]
The docs guide how to read secrets from the vault, but not how to add them.

**Comment by Mikescops** [comment]
Hello,

For now the CLI can only make read only operations. 
To add secrets you have to go through one of our GUI client.

**Comment by tsamppa** [comment]
How the hierarchy works with secure notes? Should it be JSON in the note?

**Comment by Mikescops** [comment]
You can put the content in the format, you want, check our documentation about "transformers". You also have YouTube tutorials on our channel.

https://dashlane.github.io/dashlane-cli/personal/secrets/read#transformer


https://youtube.com/playlist?list=PLaJHdXnJiAn4rWs-YlmKuoX__33BAq_xf&si=Fy5jKKUJUHweHvZX

**Comment by crossterra-agent** [comment]
- **Status**: DONE
- **Role**: orchestrator
- **Step**: 
- **Link**: https://github.com/CrossTerra/monorepo/pull/3008#issuecomment-4622724997
- **Summary**: Delta steps added for S03–S06; developer re-invoked on PR #3008.
